src/software/new_geom/vector.h

```diff
@@ -184,14 +184,26 @@ class Vector final
     Vector &operator=(const Vector &other);
 
     /**
-     * Calculate the angle between this and another Vector
+     * Calculate the min angle between this and another Vector
      *
      * @param other the other Vector to calculate the angle with
      *
      * @return the Angle between this vector and the other vector
      */
     Angle angleWith(const Vector &other) const;
 
+    /**
+     * Determines if another Vector points in a similar direction to this vector
+     *
+     * Two vectors are similar if the direction of the projection of the `other` vector
+     * onto `this` vector is the same as the direction of `this` vector
+     *
+     * @param other The other vector
+     *
+     * @return true if other is similar to this vector
+     */
+    bool similar(const Vector &other) const;
```

Comment:
This could probably benefit from a more descriptive name. I only know what it does cause I've seen this before.

Perhaps `within90Degrees` or similar, since that's the same thing as the projection definition?

@jsn0716 @scveloso @jared-paul @akhilveeraghanta @garethellis0

Reply:
This really only seems to be used in `intercept_ball_action`. One potential option could be to write this as `acuteAngle(v1, v2) < Angle::Quarter` instead of having a dedicated function for it?

Reply:
do you prefer `acuteAngle(Vector,Vector)` or `Vector::angleWith(Vector)`?

Reply:
`acuteAngle(Vector,Vector)` makes sense, (although the other one reads better), assuming we probably will have this for Rays other geom types maybe?

Reply:
changed as @garethellis0 and @akhilveeraghanta suggested